client/src/project/Form.vue

```diff
@@ -411,13 +416,73 @@
       async onSubmit() {
         const valid = await this.$validator.validateAll();
         if (valid) {
-          this.$emit('projectFormSubmitted', this.projectPayload);
+          let project;
+          if (this.projectPayload.id) {
+            project = await this.update();
+          } else {
+            project = await this.register();
+          }
+          window.removeEventListener('beforeunload', this.onBeforeUnload);
+          this.submitted = true;
+          this.$router.push({
+            name: this.event.questions && this.event.questions.length > 0 ? 'ProjectExtraDetails' : 'CreateProjectCompleted',
+            params: {
+              eventSlug: this.event.slug,
+              projectId: project.id,
+              _event: this.event,
+              _project: project,
+            },
+          });
         }
       },
+      async register() {
+        try {
+          const project = (await ProjectService.create(this.event.id, this.projectPayload)).body;
+          this.$ga.event({
+            eventCategory: 'ProjectRegistration',
+            eventAction: 'NewProject',
+            eventLabel: this.event.id,
+          });
+          return project;
+        } catch (err) {
+          this.error = err;
+        }
+      },
+      async update() {
+        try {
+          const project =
+            (await ProjectService.update(this.event.id, this.project.id, this.projectPayload)).body;
+          this.$ga.event({
+            eventCategory: 'ProjectRegistration',
+            eventAction: 'UpdateProject',
+            eventLabel: this.event.id,
+          });
+          return project;
+        } catch (err) {
+          this.error = err;
+        }
+      },
+      onBeforeUnload(e) {
+        e.returnValue = 'Are you sure you don\'t want to complete your registration application?';
+        return 'Are you sure you don\'t want to complete your registration application?';
```

Comment:
Not sure to understand why the string is set twice here, but eh

Reply:
Yeah, I could make a const and assign I guess. Would make more sense. I think this mostly came from "huh, return alone doesn't work... \*copy\* \*paste\*. Cool, it works now!"

Reply:
That's fine, no biggie, just not sure "how" is e.returnValue used differently from the return itself

Reply:
Because web standards https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload

😸

Reply:
Am i allowed to say it's a crappy standard? :D

Reply:
![](https://imgs.xkcd.com/comics/standards.png)